plugins/wysiwygarea/plugin.js

```diff
@@ -93,8 +93,44 @@
 						return;
 					}
 
-					editor.editable( new framedWysiwyg( editor, iframe.$.contentWindow.document.body ) );
+					editor.editable( new framedWysiwyg( editor, iframe.getFrameDocument().getBody() ) );
 					editor.setData( editor.getData( 1 ), callback );
+
+					iframe.$.onloadFromSetData = false;
```

Comment:
> Let's encapsulate the manipulation of this property within a method to operate on the same abstraction level as elsewhere.

I agree. Setting single `flag` properties make code hard to read and understand :+1: 

> I'd see it as a method of a `CKEDITOR.dom.element` (we have already have e.g. [`getFrameElement()`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_dom_element.html#method-getFrameDocument) method which is exclusive for iframes, so I think it's not an issue).

I wouldn't go up that much in abstraction. Property `contextWindow` [has some sense](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#dom-object-contentwindow) even for the non-iframe elements. This flag even for `iframe` itself has no meaning. Actually, it only means something in `wysiwygarea` plugin. Meaning context of this flag reaches both: plugin and class extending editable, so I would go with helper function like https://github.com/ckeditor/ckeditor4/blob/9d34608a06a79fdb490d5a3f25972ee9a4a650b5/plugins/wysiwygarea/plugin.js#L597

> So I'd suggest sth similar as for [`editor.editable()`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#method-editable) method, so if called with no parameters it returns the current value, and if called with an argument (`true` or `false`) it sets the proper value

I'm not a big fan of one method which could be both: getter and setter - but since it's already a kind of convention in existing code, it sounds like a good idea.

> However it's difficult to easily explain it :thinking:WDYT about `shouldReload()` as the name for the method I've proposed above?

Actually, this flag prevents iframe body recreation. Reload itself is called anyway. The previous proposal sounds good, because indeed - it attaches reload to iframe. In this case, the long description would be: "let's save information that `SetData` was called - because we know that it leads to another `onload` invocation". It is hard to find out a good name for one method - since it will be called in three contexts:

*   inside `SetData` : sth similar to SetDataWasCalled( true )
*   inside this so-called `backup on load function`: WasSetDataJustCalledBeforeThisOnload()
*   inside both `onload` functions as a 'clean up' after editable element creation: SetDataWasCalled ( false ).

All those above bullets lead me to sth like: `shouldRecreateEditable()` && `shouldRecreateEditable( true / false )`WDYT?

Reply:
> Meaning context of this flag reaches both: plugin and class extending editable, so I would go with helper function

Agreed 👍🏻 

> I'm not a big fan of one method which could be both: getter and setter - but since it's already a kind of convention in existing code, it sounds like a good idea.

Me neither, I wouldn't propose it in the new project, so we are on the same page here 😄 

> All those above bullets lead me to sth like: `shouldRecreateEditable( iframe )` && `shouldRecreateEditable( iframe, true / false )` WDYT?

I like it, it's informative and concise at the same time, let's go with it 👍🏻